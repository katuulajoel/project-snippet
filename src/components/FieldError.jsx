import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const FieldError = ({ message }) => {
  return (
    <Wrapper>
      {message
        ? Array.isArray(message)
          ? message.map((item, idx) => {
              return <div key={idx}>{item}</div>;
            })
          : message
        : ""}
    </Wrapper>
  );
};

FieldError.propTypes = {
  message: PropTypes.string,
};

const Wrapper = styled.div`
  color: #e93232;
`;

export default FieldError;
