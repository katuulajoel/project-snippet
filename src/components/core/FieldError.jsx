/* eslint-disable react/prop-types */
import React from "react";

const FieldError = ({ message }) => {
  return (
    <div className="error">
      {message
        ? Array.isArray(message)
          ? message.map((item, idx) => {
              return <div key={idx}>{item}</div>;
            })
          : message
        : ""}
    </div>
  );
};

export default FieldError;
