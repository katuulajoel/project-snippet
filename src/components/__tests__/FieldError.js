import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import FieldError from "../FieldError";

describe("Auth layout test", () => {
  it("Snapshot test for Login component", () => {
    const tree = renderer
      .create(
        <Router>
          <FieldError message="Error Message" />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
