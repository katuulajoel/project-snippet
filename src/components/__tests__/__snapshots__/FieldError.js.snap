// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth layout test Snapshot test for Login component 1`] = `
<div
  className="error"
>
  Error Message
</div>
`;
